packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -486,15 +516,41 @@ class _DraggableScrollableSheetScrollPosition
       final double delta = ballisticController.value - lastDelta;
       lastDelta = ballisticController.value;
       extent.addPixelDelta(delta, context.notificationContext!);
-      if ((velocity > 0 && extent.isAtMax) || (velocity < 0 && extent.isAtMin)) {
+      if ((velocity > 0 && extent.isAtMax) ||
+          (velocity < 0 && extent.isAtMin)) {
         // Make sure we pass along enough velocity to keep scrolling - otherwise
         // we just "bounce" off the top making it look like the list doesn't
         // have more to scroll.
-        velocity = ballisticController.velocity + (physics.tolerance.velocity * ballisticController.velocity.sign);
+        velocity = ballisticController.velocity +
+            (physics.tolerance.velocity * ballisticController.velocity.sign);
         super.goBallistic(velocity);
         ballisticController.stop();
       } else if (ballisticController.isCompleted) {
-        super.goBallistic(0);
+        if (extent.snapPoints.isNotEmpty) {
+          // We've stopped moving, find the closet snap point and snap to it.
+          final double targetPoint = extent.snapPoints.reduce(
+              (double a, double b) => (extent.currentExtent - a).abs() <
+                      (extent.currentExtent - b).abs()
+                  ? a
+                  : b);
+          final AnimationController snapController = AnimationController(
+            vsync: context.vsync,
+            value: extent.currentExtent,
+          );
+          CurvedAnimation(
+            parent: snapController,
+            curve: Curves.easeInOut,
+          ).addListener(() {
+            extent.updateExtent(
+                snapController.value, context.notificationContext!);
+          });
+          snapController
+              .animateTo(targetPoint,
+                  duration: const Duration(milliseconds: 200))
+              .then((_) => super.goBallistic(0));
```

Comment:
The curve and animation duration here were chosen fairly arbitrarily. Are this the right values and/or should these be user customizable?